File: week02/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);
int gradeLevel(int letters, int words, int sentences);

int main(void)
{
    // get text with get_string
    string text = get_string("Text: ");

    // count letters (a-z, A-Z)
    int letters = count_letters(text);

    //count words (any sequence of characters separated by a space)
    int words = count_words(text);

    //count sentences (marked by . ! ?)
    int sentences = count_sentences(text);

    //print Grade X (X = grade level)
    int grade = gradeLevel(letters, words, sentences);

    if (grade < 1) //if grade is <1, print before grade 1
    {
        printf("Before Grade 1\n");
    }

    else if (grade > 16) //if grade is 16+, print grade 16+
    {
        printf("Grade 16+\n");
    }

    else // otherwise print the rounded grade level
    {
        printf("Grade %i\n", grade);
    }
}

int count_letters(string text)
{
    int length = strlen(text);
    int lets = 0; // for the number of letters

    for (int i = 0; i <= length; i++)
    {
        if (isalpha(text[i])) // if the char is a letter
        {
            lets++; // count 1 letter
        }
    }

    return lets;
}

int count_words(string text)
{
    int length = strlen(text);
    int wrds = 1; //starting at 1 because the last word of a string will not end in a space

    for (int i = 0; i <= length; i++)
    {
        if (isspace(text[i])) //if char is a space
        {
            wrds++;
        }
    }

    return wrds;
}

int count_sentences(string text)
{
    int length = strlen(text);
    int sents = 0; //for counting sentences

    for (int i = 0; i <= length; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sents++;
        }
    }

    return sents;
}

int gradeLevel(int letters, int words, int sentences)
{
    float lets100 = ((float)letters / (float)words * 100.0);
    float sens100 = ((float)sentences / (float)words * 100.0);
    float index = 0.0588 * lets100 - 0.296 * sens100 - 15.8;

    return round(index);
}

File: week06/sentimental-readability/readability.py
from cs50 import get_string


def main():
    # Get text with get_string
    text = get_string("Text: ")

    # Count letters (a-z, A-Z)
    letters = count_letters(text)

    # Count words (any sequence of character separated by a space)
    words = count_words(text)

    # Count sentences (marked by . ! ?)
    sentences = count_sentences(text)

    # Print Grade X (X = Grade Level)
    grade = grade_level(letters, words, sentences)

    if grade < 1:
        print("Before Grade 1")
        return 0
    if grade > 16:
        print("Grade 16+")
    else:
        print(f"Grade {grade}")


def count_letters(text):
    count = 0  # Variable to store number of spaces
    for i in range(len(text)):
        if text[i].isalpha() == True:
            count += 1

    return count


def count_words(text):
    count = 1  # Start a 1 because the lat word of a string will not end in a space
    for i in range(len(text)):
        if text[i].isspace() == True:
            count += 1

    return count


def count_sentences(text):
    count = 0
    for i in range(len(text)):
        if text[i] == "." or text[i] == "!" or text[i] == "?":
            count += 1

    return count


def grade_level(letters, words, sentences):
    lets100 = letters / words * 100
    sens100 = sentences / words * 100
    index = 0.0588 * lets100 - 0.296 * sens100 - 15.8

    return round(index)


if __name__ == "__main__":
    main()